tests/Validators/PluginOrderValidatorTest.php

```diff
@@ -0,0 +1,226 @@
+
+namespace Findologic\Tests\Validators;
+
+use Findologic\Validators\PluginOrderValidator;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Plenty\Modules\Plugin\Contracts\PluginRepositoryContract;
+use Plenty\Modules\Plugin\Models\Plugin;
+use Plenty\Modules\Plugin\PluginSet\Contracts\PluginSetRepositoryContract;
+
+class PluginOrderValidatorTest extends TestCase
+{
+    /**
+     * @var PluginOrderValidator|MockObject
+     */
+    private $validatorMock;
+
+    /**
+     * @var PluginSetRepositoryContract|MockObject
+     */
+    private $pluginSetRepositoryMock;
+
+    /**
+     * @var PluginRepositoryContract|MockObject
+     */
+    private $pluginRepositoryMock;
+
+    public function setup()
+    {
+        /** @var PluginSetRepositoryContract|MockObject $pluginSetRepositoryMock */
+        $this->pluginSetRepositoryMock = $this->getMockBuilder(PluginSetRepositoryContract::class)
+            ->setMethods(['getCurrentPluginSetId'])
+            ->getMockForAbstractClass();
+
+        /** @var PluginRepositoryContract|MockObject $pluginRepositoryMock */
+        $this->pluginRepositoryMock = $this->getMockBuilder(PluginRepositoryContract::class)
+            ->setMethods(['getCurrentPluginSetId'])
+            ->getMockForAbstractClass();
+
+        /** @var PluginOrderValidator|MockObject $validatorMock */
+        $this->validatorMock = $this->getMockBuilder(PluginOrderValidator::class)
+            ->setMethods(['getPluginRepository', 'getPluginSetRepository'])
+            ->getMock();
+        $this->validatorMock->method('getPluginSetRepository')->willReturn($this->pluginSetRepositoryMock);
+        $this->validatorMock->method('getPluginRepository')->willReturn($this->pluginRepositoryMock);
+    }
+
+    /**
+     * @dataProvider validationTestDataProvider
+     * @param array $ioPluginSetEntries
+     * @param array $findologicPluginSetEntries
+     * @param bool $expectedResult
+     */
+    public function testValidation($ioPluginSetEntries, $findologicPluginSetEntries, $expectedResult)
+    {
+        $this->pluginSetRepositoryMock->method('getCurrentPluginSetId')->willReturn('1');
+
+        $ioPluginMock = $this->getMockForAbstractClass(Plugin::class);
+        $ioPluginMock->pluginSetEntries = $ioPluginSetEntries;
+
+        $findologicPluginMock = $this->getMockForAbstractClass(Plugin::class);
+        $findologicPluginMock->pluginSetEntries = $findologicPluginSetEntries;
+
+        $this->pluginRepositoryMock->method('getPluginByName')
+            ->withConsecutive(['io'], ['findologic'])
+            ->willReturnOnConsecutiveCalls($ioPluginMock, $findologicPluginMock);
+
+        $this->assertEquals($expectedResult, $this->validatorMock->validate());
+    }
+
+    public function validationTestDataProvider(): array
+    {
+        return [
+            'It is valid when IO is loaded before Findologic in the current plugin set' => [
+                [
+                    (object) [
```

Comment:
Why are these casted into raw objects? You could use a mock, which is created inside of the test instead.

Reply:
For convenience and saving time while writing these tests. Sure, I could use a mock, but would it be somehow better in any way?

Reply:
Nevermind, found the type. Ok, I'll make these into mocks then

Reply:
It is overall easier to change, and I personally prefer mocks over some random objects floating around.